src/snapred/backend/recipe/algorithm/CalibrationNormalizationAlgo.py

```diff
@@ -0,0 +1,149 @@
+        ingredients = ingredients(**json.loads(self.getProperty("Ingredients").value))
+        self.chopIngredients(ingredients)
+
+        focusGroups = self.reductionIngredients.reductionState.stateConfig.focusGroups.json()
+        runNumber: str = self.run.runNumber
+        ipts: str = self.run.IPTS
+        rawDataPath: str = ipts + "shared/lite/SNAP_{}.lite.nxs.h5".format(runNumber)
+        backgroundRunNum: str = self.backgroundRun.runNumber
+        backgroundIpts: str = self.backgroundRun.IPTS
+        backgroundRawDataPath: str = backgroundIpts + "shared/lite/SNAP_{}.lite.nxs.h5".format(backgroundRunNum)
+
+        inputWS = self.getProperty("InputWorkspace").value
+        inputWSName = str(inputWS)
+        if inputWS is None:
+            self.mantidSnapper.Load(
+                "Loading file for Normalization InputWS...",
+                Filename=rawDataPath,
+                OutputWorkspace=inputWS,
+            )
+        backgroundWS = self.getProperty("BackgroundWorkspace").value
+        if backgroundWS is None:
+            self.mantidSnapper.Load(
+                "Loading file for Normalization BackgroundWS...",
+                Filename=backgroundRawDataPath,
+                OutputWorkspace=backgroundWS,
+            )
+        # run the algo
+        self.log().notice("Execution of CalibrationNormalizationAlgo START!")
+
+        raw_data = self.mantidSnapper.RawVanadiumCorrectionAlgorithm(
+            "Correcting Vanadium Data...",
+            InputWorkspace=inputWS,
+            BackgroundWorkspace=backgroundWS,
+            CalibrationWorkspace=self.calibrationWorkspace.json(),
+            Ingredients=self.reductionIngredients.json(),
+            CalibrantSample=self.calibrantSample.json(),
+            OutputWorkspace=f"{inputWSName}_raw_data",
+        )
+
+        groupingworkspace = self.mantidSnapper.CustomGroupWorkspace(
+            "Creating Group Workspace...",
+            StateConfig=self.reductionIngredients.reductionState.stateConfig.json(),
+            InputWorkspace=raw_data,
+            OutputWorkspace="CommonRed",
+        )
+
+        for workspaceIndex in range(len(focusGroups)):
+            data = self.mantidSnapper.RebinRagged(
+                "Rebinning ragged bins...",
+                InputWorkspace=mtd[raw_data].getItem(workspaceIndex),
+                XMin=focusGroups[workspaceIndex].dMin,
+                XMax=focusGroups[workspaceIndex].dMax,
+                Delta=focusGroups[workspaceIndex].dBin,
+                OutputWorkspace="data_rebinned_ragged_" + str(focusGroups[workspaceIndex].name),
+                PreserveEvents=False,
+            )
+
+        focused_data = self.mantidSnapper.DiffractionFocussing(  # GroupDetectors instead
+            "Performing Diffraction Focusing ...",
+            InputWorkspace=data,
+            GroupingWorkspace=groupingworkspace,
+            OutputWorkspace=f"{inputWSName}focused_data",
+            PreserveEvents=True,
+        )
+
```

Comment:
This is where `RebinRagged` should occur and it should operate on  `OutputWorkspace=f"{inputWSName}focused_data"`